### DISCUSSION TITLE
Prisma Client doesn't support for OpenSSL

### DISCUSSION THREAD

**Discussion by HendrixNguyen** [prisma/prisma#20247] [Q&A]
Hi everyone,
I use arch Linux, and when I install openssl that my openssl version is at 3.1.x. Unluckyly that prisma client is only support for openssl@3.0.x
![image](https://github.com/prisma/prisma/assets/48069618/53fd2fb7-5356-4ade-866f-24abef05cbd4)

Do we have any idea for this problem ?

**Comment by masterbater** [comment]
It should but your os might not be, just native to autodetect

**Comment by ludralph** [comment]
Hi @HendrixNguyen 

You would need to specify one of the following supported OpenSSL versions in the [binaryTargets](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#binarytargets-options) option. You could also try using the `native` option to see if it resolves the error.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by janpio** [comment]
That sounds wrong @HendrixNguyen, Prisma should not detect this as `openssl-3.1.x`.

What Prisma version are you using?
You are using the Javascript/Typescript Client, correct?
What operating system (exactly) are you using?
How exactly did you configure your project? Are you using `PRISMA_QUERY_ENGINE_LIBRARY` manually?

**Comment by abenbouya** [comment]
Anyone found a solution for this ?
